### ISSUE TITLE
Build fails if Vulkan SDK > 1.3.216.0 is present on the system - Windows / MSYS2

### ISSUE THREAD

**Issue by Chipcraft** [haasn/libplacebo#153]
Commit: 72b2c020150880041d32b42071ea7d3e517d6873

Configure: "meson setup --prefix=/home/Chipcraft/FFLib --buildtype=release Buildi"

Compile: "ninja -CBuildi install"

With 1.3.216.0 SDK installed everything works perfectly fine however, with any of the newer SDKs (224, 231, 236, 239 - https://vulkan.lunarg.com/sdk/home) present on the system:

```
ninja: Entering directory `Buildi'
[2/71] Generating src/vulkan/utils_gen.c with a custom command (wrapped by meson to set env)
Found vk.xml: C:\VulkanSDK\1.3.224.1/share/vulkan/registry/vk.xml
[4/71] Generating src/opengl/include/glad/gl.h with a custom command (wrapped by meson to set env)
[08.02.2023 19:48:43][INFO      ][glad          ]: getting 'gl' specification from remote location
[08.02.2023 19:48:43][DEBUG     ][glad.files    ]: intercepted attempt to retrieve resource: 'https:
//raw.githubusercontent.com/KhronosGroup/OpenGL-Registry/main/xml/gl.xml'
[08.02.2023 19:48:43][INFO      ][glad.files    ]: opening packaged resource: 'gl.xml'
[08.02.2023 19:48:43][INFO      ][glad          ]: getting 'egl' specification from remote location
[08.02.2023 19:48:43][DEBUG     ][glad.files    ]: intercepted attempt to retrieve resource: 'https:
//raw.githubusercontent.com/KhronosGroup/EGL-Registry/main/api/egl.xml'
[08.02.2023 19:48:43][INFO      ][glad.files    ]: opening packaged resource: 'egl.xml'
[08.02.2023 19:48:43][INFO      ][glad          ]: generating gl:core/gl=None
[08.02.2023 19:48:43][INFO      ][glad          ]: no explicit version given for api gl, using Versi
on(major=4, minor=6)
[08.02.2023 19:48:43][INFO      ][glad          ]: generating gles2:None/gl=None
[08.02.2023 19:48:43][INFO      ][glad          ]: no explicit version given for api gles2, using Ve
rsion(major=3, minor=2)
[08.02.2023 19:48:43][INFO      ][glad          ]: merging [FeatureSet(name=gl, info=[gl:core=4.6],
extensions=20), FeatureSet(name=gles2, info=[gles2=3.2], extensions=10)]
[08.02.2023 19:48:43][INFO      ][glad          ]: merged into FeatureSet(name=gl, info=[gl:core=4.6
, gles2=3.2], extensions=28)
[08.02.2023 19:48:43][INFO      ][glad          ]: generating feature set FeatureSet(name=gl, info=[
gl:core=4.6, gles2=3.2], extensions=28)
[08.02.2023 19:48:43][DEBUG     ][glad.files    ]: intercepted attempt to retrieve resource: 'https:
//raw.githubusercontent.com/KhronosGroup/EGL-Registry/main/api/KHR/khrplatform.h'
[08.02.2023 19:48:43][INFO      ][glad.files    ]: opening packaged resource: 'khrplatform.h'
[08.02.2023 19:48:44][INFO      ][glad          ]: generating egl:None/egl=None
[08.02.2023 19:48:44][INFO      ][glad          ]: no explicit version given for api egl, using Vers
ion(major=1, minor=5)
[08.02.2023 19:48:44][INFO      ][glad          ]: merging [FeatureSet(name=egl, info=[egl=1.5], ext
ensions=7)]
[08.02.2023 19:48:44][INFO      ][glad          ]: merged into FeatureSet(name=egl, info=[egl=1.5],
extensions=7)
[08.02.2023 19:48:44][INFO      ][glad          ]: generating feature set FeatureSet(name=egl, info=
[egl=1.5], extensions=7)
[08.02.2023 19:48:44][DEBUG     ][glad.files    ]: intercepted attempt to retrieve resource: 'https:
//raw.githubusercontent.com/KhronosGroup/EGL-Registry/main/api/EGL/eglplatform.h'
[08.02.2023 19:48:44][INFO      ][glad.files    ]: opening packaged resource: 'eglplatform.h'
[62/71] Compiling C object src/libplacebo-241.dll.p/meson-generated_.._vulkan_utils_gen.c.obj
FAILED: src/libplacebo-241.dll.p/meson-generated_.._vulkan_utils_gen.c.obj
"cc" "-Isrc/libplacebo-241.dll.p" "-Isrc" "-I../src" "-Isrc/include" "-I../src/include" "-Isrc/openg
l/include" "-I../src/opengl/include" "-Isrc/vulkan" "-Isrc/opengl/include/glad" "-ID:/msys64/mingw64
/include/spirv_cross" "-fdiagnostics-color=always" "-D_FILE_OFFSET_BITS=64" "-Wall" "-Winvalid-pch"
"-Wextra" "-std=c11" "-O3" "-Wundef" "-Wshadow" "-Wparentheses" "-Wpointer-arith" "-D_WIN32_WINNT=0x
0601" "-D_ISOC99_SOURCE" "-D_ISOC11_SOURCE" "-D_GNU_SOURCE" "-D_XOPEN_SOURCE=700" "-U__STRICT_ANSI__
" "-fvisibility=hidden" "-Wmissing-prototypes" "-Wno-sign-compare" "-Wno-unused-parameter" "-Wno-mis
sing-field-initializers" "-Wno-type-limits" "-Werror=implicit-function-declaration" "-Werror=incompa
tible-pointer-types" "-DCOBJMACROS" "-DINITGUID" -MD -MQ src/libplacebo-241.dll.p/meson-generated_..
_vulkan_utils_gen.c.obj -MF "src/libplacebo-241.dll.p/meson-generated_.._vulkan_utils_gen.c.obj.d" -
o src/libplacebo-241.dll.p/meson-generated_.._vulkan_utils_gen.c.obj "-c" src/vulkan/utils_gen.c
src/vulkan/utils_gen.c: In function 'vk_struct_size':
src/vulkan/utils_gen.c:1045:10: error: 'VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_AMIGO_PROFILING_FEATURES_S
EC' undeclared (first use in this function); did you mean 'VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_P
ROCESSING_FEATURES_QCOM'?
 1045 |     case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_AMIGO_PROFILING_FEATURES_SEC: return sizeof(VkPhy
sicalDeviceAmigoProfilingFeaturesSEC);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |          VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_PROCESSING_FEATURES_QCOM
src/vulkan/utils_gen.c:1045:10: note: each undeclared identifier is reported only once for each func
tion it appears in
src/vulkan/utils_gen.c:1045:88: error: 'VkPhysicalDeviceAmigoProfilingFeaturesSEC' undeclared (first
 use in this function); did you mean 'VkPhysicalDeviceImageProcessingFeaturesQCOM'?
 1045 |     case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_AMIGO_PROFILING_FEATURES_SEC: return sizeof(VkPhy
sicalDeviceAmigoProfilingFeaturesSEC);
      |                                                                                        ^~~~~
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                        VkPhy
sicalDeviceImageProcessingFeaturesQCOM
src/vulkan/utils_gen.c:1046:10: error: 'VK_STRUCTURE_TYPE_AMIGO_PROFILING_SUBMIT_INFO_SEC' undeclare
d (first use in this function); did you mean 'VK_STRUCTURE_TYPE_ACQUIRE_PROFILING_LOCK_INFO_KHR'?
 1046 |     case VK_STRUCTURE_TYPE_AMIGO_PROFILING_SUBMIT_INFO_SEC: return sizeof(VkAmigoProfilingSu
bmitInfoSEC);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |          VK_STRUCTURE_TYPE_ACQUIRE_PROFILING_LOCK_INFO_KHR
src/vulkan/utils_gen.c:1046:75: error: 'VkAmigoProfilingSubmitInfoSEC' undeclared (first use in this
 function)
 1046 |     case VK_STRUCTURE_TYPE_AMIGO_PROFILING_SUBMIT_INFO_SEC: return sizeof(VkAmigoProfilingSu
bmitInfoSEC);
      |                                                                           ^~~~~~~~~~~~~~~~~~
~~~~~~~~~~~
src/vulkan/utils_gen.c:1047:10: error: 'VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_ATTACHMENT_FEEDBACK_LOOP_L
AYOUT_FEATURES_EXT' undeclared (first use in this function); did you mean 'VK_STRUCTURE_TYPE_PHYSICA
L_DEVICE_SCALAR_BLOCK_LAYOUT_FEATURES_EXT'?
 1047 |     case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_ATTACHMENT_FEEDBACK_LOOP_LAYOUT_FEATURES_EXT: ret
urn sizeof(VkPhysicalDeviceAttachmentFeedbackLoopLayoutFeaturesEXT);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |          VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SCALAR_BLOCK_LAYOUT_FEATURES_EXT
src/vulkan/utils_gen.c:1047:104: error: 'VkPhysicalDeviceAttachmentFeedbackLoopLayoutFeaturesEXT' un
declared (first use in this function); did you mean 'VkPhysicalDeviceScalarBlockLayoutFeaturesEXT'?
 1047 |     case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_ATTACHMENT_FEEDBACK_LOOP_LAYOUT_FEATURES_EXT: ret
urn sizeof(VkPhysicalDeviceAttachmentFeedbackLoopLayoutFeaturesEXT);
      |
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |
           VkPhysicalDeviceScalarBlockLayoutFeaturesEXT
[66/71] Compiling C object demos/libnuklear.a.p/ui.c.obj
ninja: build stopped: subcommand failed.

```

**Comment by haasn** [comment]
Are you possibly including `vulkan.h` from a different (perhaps system-wide) location than the `vk.xml`?

**Comment by Chipcraft** [comment]
MSYS2 does have Vulkan headers present without any of the actual Vulkan packages installed however, not sure what to do about it when it clearly uses the correct vk.xml registry path (the external SDK).

**Comment by haasn** [comment]
The problem is a mismatch between the path it pulls `vk.xml` from and the path it pulls `vulkan.h` from. I'm not sure what we can do to work around this..
